insert_by_period/macros/get_period_boundaries.sql

```diff
@@ -1,13 +1,17 @@
-{% macro get_period_boundaries(target_schema, target_table, timestamp_field, start_date, stop_date, period) -%}
-    {{ return(adapter.dispatch('get_period_boundaries', 'insert_by_period')(target_schema, target_table, timestamp_field, start_date, stop_date, period)) }}
+{% macro get_period_boundaries(target_schema, target_table, timestamp_field, start_date, stop_date, period, overwrite, full_refresh_mode) -%}
+    {{ return(adapter.dispatch('get_period_boundaries', 'insert_by_period')(target_schema, target_table, timestamp_field, start_date, stop_date, period, overwrite, full_refresh_mode)) }}
 {% endmacro %}
 
-{% macro default__get_period_boundaries(target_schema, target_table, timestamp_field, start_date, stop_date, period) -%}
+{% macro default__get_period_boundaries(target_schema, target_table, timestamp_field, start_date, stop_date, period, overwrite, full_refresh_mode) -%}
 
   {% call statement('period_boundaries', fetch_result=True) -%}
     with data as (
       select
-          coalesce(max({{timestamp_field}}), '{{start_date}}')::timestamp as start_timestamp,
+          {% if overwrite and not full_refresh_mode -%}
+            '{{start_date}}'::timestamp as start_timestamp,
+          {%- else -%}
+            coalesce(max({{timestamp_field}}), '{{start_date}}')::timestamp as start_timestamp,
+          {%- endif %}
```

Comment:
If we're already updating this, how would you feel about using the `cast` function instead of `::`?

The former works across more data platforms.

Also: v1.8 adds a [cross-database `cast` macro](https://github.com/dbt-labs/dbt-core/issues/9976), so we can use that at some point too.

Reply:
`CAST()` makes sense - will update.